Antrea Controller can no longer run on 32-bit architectures

**Describe the bug**
The 32-bit ARM CI job has been failing ever since https://github.com/antrea-io/antrea/commit/7faeaaf54ff06c4ea33399d34f6d5c208b1c3bcf was merged:

```
E0714 19:35:21.612048       1 runtime.go:78] Observed a panic: "invalid memory address or nil pointer dereference" (runtime error: invalid memory address or nil pointer dereference)
goroutine 346 [running]:
k8s.io/apimachinery/pkg/util/runtime.logPanic(0x1675a50, 0x276c988)
	/go/pkg/mod/k8s.io/apimachinery@v0.21.0/pkg/util/runtime/runtime.go:74 +0x74
k8s.io/apimachinery/pkg/util/runtime.HandleCrash(0x0, 0x0, 0x0)
	/go/pkg/mod/k8s.io/apimachinery@v0.21.0/pkg/util/runtime/runtime.go:48 +0x6c
panic(0x1675a50, 0x276c988)
	/usr/local/go/src/runtime/panic.go:969 +0x158
runtime/internal/atomic.goXadd64(0x38da44c, 0x1, 0x0, 0x4620000, 0x0)
	/usr/local/go/src/runtime/internal/atomic/atomic_arm.go:103 +0x1c
antrea.io/antrea/pkg/controller/grouping.(*GroupEntityController).addPod(0x38da440, 0x1864580, 0x4620000)
	/antrea/pkg/controller/grouping/controller.go:151 +0x144
k8s.io/client-go/tools/cache.ResourceEventHandlerFuncs.OnAdd(...)
	/go/pkg/mod/k8s.io/client-go@v0.21.0/tools/cache/controller.go:231
k8s.io/client-go/tools/cache.(*processorListener).run.func1()
	/go/pkg/mod/k8s.io/client-go@v0.21.0/tools/cache/shared_informer.go:777 +0xc0
k8s.io/apimachinery/pkg/util/wait.BackoffUntil.func1(0x3ce77bc)
	/go/pkg/mod/k8s.io/apimachinery@v0.21.0/pkg/util/wait/wait.go:155 +0x54
k8s.io/apimachinery/pkg/util/wait.BackoffUntil(0x4639fbc, 0x1b35d30, 0x3eb55a0, 0x15da901, 0x3e3c6c0)
	/go/pkg/mod/k8s.io/apimachinery@v0.21.0/pkg/util/wait/wait.go:156 +0x74
k8s.io/apimachinery/pkg/util/wait.JitterUntil(0x3ce77bc, 0x3b9aca00, 0x0, 0x0, 0x0, 0x1, 0x3e3c6c0)
	/go/pkg/mod/k8s.io/apimachinery@v0.21.0/pkg/util/wait/wait.go:133 +0x80
k8s.io/apimachinery/pkg/util/wait.Until(...)
	/go/pkg/mod/k8s.io/apimachinery@v0.21.0/pkg/util/wait/wait.go:90
k8s.io/client-go/tools/cache.(*processorListener).run(0x38ef1a0)
	/go/pkg/mod/k8s.io/client-go@v0.21.0/tools/cache/shared_informer.go:771 +0x88
k8s.io/apimachinery/pkg/util/wait.(*Group).Start.func1(0x3cbe62c, 0x406e118)
	/go/pkg/mod/k8s.io/apimachinery@v0.21.0/pkg/util/wait/wait.go:73 +0x4c
created by k8s.io/apimachinery/pkg/util/wait.(*Group).Start
	/go/pkg/mod/k8s.io/apimachinery@v0.21.0/pkg/util/wait/wait.go:71 +0x4c
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x12360]

goroutine 346 [running]:
k8s.io/apimachinery/pkg/util/runtime.HandleCrash(0x0, 0x0, 0x0)
	/go/pkg/mod/k8s.io/apimachinery@v0.21.0/pkg/util/runtime/runtime.go:55 +0xec
panic(0x1675a50, 0x276c988)
	/usr/local/go/src/runtime/panic.go:969 +0x158
runtime/internal/atomic.goXadd64(0x38da44c, 0x1, 0x0, 0x4620000, 0x0)
	/usr/local/go/src/runtime/internal/atomic/atomic_arm.go:103 +0x1c
antrea.io/antrea/pkg/controller/grouping.(*GroupEntityController).addPod(0x38da440, 0x1864580, 0x4620000)
	/antrea/pkg/controller/grouping/controller.go:151 +0x144
k8s.io/client-go/tools/cache.ResourceEventHandlerFuncs.OnAdd(...)
	/go/pkg/mod/k8s.io/client-go@v0.21.0/tools/cache/controller.go:231
k8s.io/client-go/tools/cache.(*processorListener).run.func1()
	/go/pkg/mod/k8s.io/client-go@v0.21.0/tools/cache/shared_informer.go:777 +0xc0
k8s.io/apimachinery/pkg/util/wait.BackoffUntil.func1(0x3ce77bc)
	/go/pkg/mod/k8s.io/apimachinery@v0.21.0/pkg/util/wait/wait.go:155 +0x54
k8s.io/apimachinery/pkg/util/wait.BackoffUntil(0x4639fbc, 0x1b35d30, 0x3eb55a0, 0x15da901, 0x3e3c6c0)
	/go/pkg/mod/k8s.io/apimachinery@v0.21.0/pkg/util/wait/wait.go:156 +0x74
k8s.io/apimachinery/pkg/util/wait.JitterUntil(0x3ce77bc, 0x3b9aca00, 0x0, 0x0, 0x0, 0x1, 0x3e3c6c0)
	/go/pkg/mod/k8s.io/apimachinery@v0.21.0/pkg/util/wait/wait.go:133 +0x80
k8s.io/apimachinery/pkg/util/wait.Until(...)
	/go/pkg/mod/k8s.io/apimachinery@v0.21.0/pkg/util/wait/wait.go:90
k8s.io/client-go/tools/cache.(*processorListener).run(0x38ef1a0)
	/go/pkg/mod/k8s.io/client-go@v0.21.0/tools/cache/shared_informer.go:771 +0x88
k8s.io/apimachinery/pkg/util/wait.(*Group).Start.func1(0x3cbe62c, 0x406e118)
	/go/pkg/mod/k8s.io/apimachinery@v0.21.0/pkg/util/wait/wait.go:73 +0x4c
created by k8s.io/apimachinery/pkg/util/wait.(*Group).Start
	/go/pkg/mod/k8s.io/apimachinery@v0.21.0/pkg/util/wait/wait.go:71 +0x4c
```

This is because of our use of `atomic.LoadUint64`. As per the Go [documentation](https://pkg.go.dev/sync/atomic#pkg-note-BUG) for the `sync/atomic` package:

> On ARM, 386, and 32-bit MIPS, it is the caller's responsibility to arrange for 64-bit alignment of 64-bit words accessed atomically. The first word in a variable or in an allocated struct, array, or slice can be relied upon to be 64-bit aligned.

`podAddEvents`, `externalEntityAddEvents` and `namespaceAddEvents` are declared in the middle of a struct and therefore have no alignment guarantee:
https://github.com/antrea-io/antrea/blob/5c95cb0d12506a9000f630cbf29a60c6b58981fd/pkg/controller/grouping/controller.go#L38-L58

**Versions:**
Antrea ToT, Antrea v1.2.0

--- a/pkg/controller/grouping/controller.go
+++ b/pkg/controller/grouping/controller.go
@@ -35,24 +35,45 @@ const (
 	resyncPeriod time.Duration = 0
 )
 
+// eventsCounter is used to keep track of the number of occurrences of an event type. It uses the
+// low-level atomic memory primitives from the sync/atomic package to provide atomic operations
+// (Increment and Load).
+// There is a known-bug on 32-bit architectures for sync/atomic:
+// On ARM, 386, and 32-bit MIPS, it is the caller's responsibility to arrange for 64-bit alignment
+// of 64-bit words accessed atomically. The first word in a variable or in an allocated struct,
+// array, or slice can be relied upon to be 64-bit aligned.
+// As a result, instances of eventsCounter should be allocated when using them in structs; they
+// should not be embedded directly.
+type eventsCounter struct {
+	count uint64
+}
+
+func (c *eventsCounter) Increment() {
+	atomic.AddUint64(&c.count, 1)
+}
+
+func (c *eventsCounter) Load() uint64 {
+	return atomic.LoadUint64(&c.count)
+}
+
 type GroupEntityController struct {
 	podInformer coreinformers.PodInformer
 	// podListerSynced is a function which returns true if the Pod shared informer has been synced at least once.
 	podListerSynced cache.InformerSynced
-	// podAddEvents is the number of Pod Add events that have been processed.
-	podAddEvents uint64
+	// podAddEvents tracks the number of Pod Add events that have been processed.
+	podAddEvents *eventsCounter
 
 	externalEntityInformer crdv1a2informers.ExternalEntityInformer
 	// externalEntityListerSynced is a function which returns true if the ExternalEntity shared informer has been synced at least once.
 	externalEntityListerSynced cache.InformerSynced
-	// externalEntityAddEvents is the number of ExternalEntity Add events that have been processed.
-	externalEntityAddEvents uint64
+	// externalEntityAddEvents tracks the number of ExternalEntity Add events that have been processed.
+	externalEntityAddEvents *eventsCounter
 
 	namespaceInformer coreinformers.NamespaceInformer
 	// namespaceListerSynced is a function which returns true if the Namespace shared informer has been synced at least once.
 	namespaceListerSynced cache.InformerSynced
-	// namespaceAddEvents is the number of Namespace Add events that have been processed.
-	namespaceAddEvents uint64
+	// namespaceAddEvents tracks the number of Namespace Add events that have been processed.
+	namespaceAddEvents *eventsCounter
 
 	groupEntityIndex *GroupEntityIndex
 }
@@ -65,10 +86,13 @@ func NewGroupEntityController(groupEntityIndex *GroupEntityIndex,
 		groupEntityIndex:           groupEntityIndex,
 		podInformer:                podInformer,
 		podListerSynced:            podInformer.Informer().HasSynced,
+		podAddEvents:               new(eventsCounter),
 		namespaceInformer:          namespaceInformer,
 		namespaceListerSynced:      namespaceInformer.Informer().HasSynced,
+		namespaceAddEvents:         new(eventsCounter),
 		externalEntityInformer:     externalEntityInformer,
 		externalEntityListerSynced: externalEntityInformer.Informer().HasSynced,
+		externalEntityAddEvents:    new(eventsCounter),
 	}
 	// Add handlers for Pod events.
 	podInformer.Informer().AddEventHandlerWithResyncPeriod(
@@ -125,14 +149,14 @@ func (c *GroupEntityController) Run(stopCh <-chan struct{}) {
 
 	// Wait until all event handlers process the initial resources before setting groupEntityIndex as synced.
 	if err := wait.PollImmediateUntil(100*time.Millisecond, func() (done bool, err error) {
-		if uint64(initialPodCount) > atomic.LoadUint64(&c.podAddEvents) {
+		if uint64(initialPodCount) > c.podAddEvents.Load() {
 			return false, nil
 		}
-		if uint64(initialNamespaceCount) > atomic.LoadUint64(&c.namespaceAddEvents) {
+		if uint64(initialNamespaceCount) > c.namespaceAddEvents.Load() {
 			return false, nil
 		}
 		if features.DefaultFeatureGate.Enabled(features.AntreaPolicy) {
-			if uint64(initialExternalEntityCount) > atomic.LoadUint64(&c.externalEntityAddEvents) {
+			if uint64(initialExternalEntityCount) > c.externalEntityAddEvents.Load() {
 				return false, nil
 			}
 		}
@@ -148,7 +172,7 @@ func (c *GroupEntityController) addPod(obj interface{}) {
 	pod := obj.(*v1.Pod)
 	klog.V(2).Infof("Processing Pod %s/%s ADD event, labels: %v", pod.Namespace, pod.Name, pod.Labels)
 	c.groupEntityIndex.AddPod(pod)
-	atomic.AddUint64(&c.podAddEvents, 1)
+	c.podAddEvents.Increment()
 }
 
 func (c *GroupEntityController) updatePod(_, curObj interface{}) {
@@ -178,7 +202,7 @@ func (c *GroupEntityController) addNamespace(obj interface{}) {
 	namespace := obj.(*v1.Namespace)
 	klog.V(2).Infof("Processing Namespace %s ADD event, labels: %v", namespace.Name, namespace.Labels)
 	c.groupEntityIndex.AddNamespace(namespace)
-	atomic.AddUint64(&c.namespaceAddEvents, 1)
+	c.namespaceAddEvents.Increment()
 }
 
 func (c *GroupEntityController) updateNamespace(_, curObj interface{}) {
@@ -209,7 +233,7 @@ func (c *GroupEntityController) addExternalEntity(obj interface{}) {
 	ee := obj.(*v1alpha2.ExternalEntity)
 	klog.V(2).Infof("Processing ExternalEntity %s/%s ADD event, labels: %v", ee.GetNamespace(), ee.GetName(), ee.GetLabels())
 	c.groupEntityIndex.AddExternalEntity(ee)
-	atomic.AddUint64(&c.externalEntityAddEvents, 1)
+	c.externalEntityAddEvents.Increment()
 }
 
 func (c *GroupEntityController) updateExternalEntity(_, curObj interface{}) {

